packages/filecoin-api/src/aggregator/buffer-reducing.js

```diff
@@ -232,9 +232,15 @@ export async function getBufferedPieces(bufferPieces, bufferStore) {
   let bufferedPieces = []
   for (const b of getBufferRes) {
     if (b.error) return b
-    bufferedPieces = bufferedPieces.concat(b.ok.buffer.pieces || [])
+    for (const piece of b.ok.buffer.pieces) {
+      const isDuplicate = bufferedPieces.some(
+        existingPiece => existingPiece.piece.equals(piece.piece)
```

Comment:
Maybe add to a `Set` and then use `.has`? - I think that might be quicker/less work - `.equals` does full buffer comparison each time.

Reply:
good call, I think it should be faster indeed